### PR TITLE
switch to blacklight startover button component

### LINKED ISSUE #2557: start_over_path is deprecated
```
DEPRECATION WARNING: start_over_path is deprecated and will be removed from a future release (Removed without replacement). (called from _app_views_catalog__start_over_html_erb___1659957843188787546_126400 at /opt/app/exhibits/exhibits/releases/20240916110307/app/views/catalog/_start_over.html.erb:1)
```

### PR DESCRIPTION
**Pull request by dnoneill**
closes #2557

### CODE REVIEW

File: app/components/blacklight/start_over_button_component.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Blacklight
```
Comment (jcoyne) [inline app/components/blacklight/start_over_button_component.rb:3]: Please don't use the namespace already in use in Blacklight (`Blacklight::StartOverButtonComponent`) . This is confusing. Instead, define `StartOverButtonComponent` at the root level (and extend `Blacklight::StartOverButtonComponent`)

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/components/blacklight/start_over_button_component.rb (hunk 1 above)
Comment (dnoneill) [reply app/components/blacklight/start_over_button_component.rb:3]: If you don't use the blacklight namespace you can't override the StartOverButton in Blacklight.

Comment (jcoyne) [reply app/components/blacklight/start_over_button_component.rb:3]: I don't think you need to override it.  Make your own class.  call the new class in `app/views/catalog/_start_over.html.erb` This makes it very explicit about what you are doing.

Comment (dnoneill) [reply app/components/blacklight/start_over_button_component.rb:3]: Then we have to also override the constraints component as well. https://github.com/projectblacklight/blacklight/blob/d566efebb088ef87bffeb5a6b5a12c5017adda5f/app/components/blacklight/constraints_component.rb#L28

Comment (jcoyne) [reply app/components/blacklight/start_over_button_component.rb:3]: @dnoneill you can pass that class as an argument to ContstraintsComponent.  No overriding is required.

Comment (dnoneill) [reply app/components/blacklight/start_over_button_component.rb]: Thanks for the tip. Updated.

**Review by jcoyne — APPROVED** [review]

**Pull request merged**